engine/src/main/java/org/hibernate/validator/constraints/br/CNPJ.java

```diff
@@ -31,7 +31,7 @@
  *
  * @author George Gastaldi
  */
-@Pattern(regexp = "([0-9]{2}[.]?[0-9]{3}[.]?[0-9]{3}[/]?[0-9]{4}[-]?[0-9]{2})")
+@Pattern(regexp = "([0-9A-Z]{2}[.]?[0-9A-Z]{3}[.]?[0-9A-Z]{3}[/]?[0-9A-Z]{4}[-]?[0-9]{2})")
```

Comment:
Since this pattern now depends on the CNPJ's version/format, I'd suggest moving it into the CNPJ validator itself and picking the pattern that suits the format.
Doing so would also mean that `@ReportAsSingleViolation`  can be removed.